### ISSUE TITLE
add tide data for secondary ports

### ISSUE THREAD

**Issue by robinpaulson** [jevonlongdell/nztides#6]
The linz website provides methods to calculate tides at a number of other harbours, e.g. mahurangi, where I was boating this week. It'd be great to figure out the data and add to this app. I'm willing to assist with the calculations, would appreciate someone helping, by checking them afterwards

**Comment by jevonlongdell** [comment]
I don't have a lot of time to look work on this but it sounds like a good idea. Checking the calculations is something I can easily do. It would make the program quite a bit more complicated.

**Comment by robinpaulson** [comment]
Oh, why would it make the programme more complicated, I imagined merely adding more .tdat files? It will certainly be much bigger though.

I'll get on with the calculations, although it appears there are a few secondary ports with no accurate data for time and/or depth.

**Comment by jevonlongdell** [comment]
I thought it would make the program more complicated in two ways
- By adding more ports it will make port selection more complicated. There are for example two Waitangi's the one everyone knows about and the main settlement on the chathams.
- By having the app have to do the calculations (although you might have been thinking along different lines) If you were just going to give times this would be quite easy, if you wanted to do heights you would have to worry about when the spring tides occured etc.

**Comment by robinpaulson** [comment]
Yes, it will make port selection more difficult, I already got caught out by "French Bay", apparently it's not the French Bay near Tititrangi....
There are a few things we could consider to make this better I'm sure, I will take a look at that and come up with some suggestions once the data is ready. For instance, perhaps we list them geographically, in groups, rather than alphabetically?

My thought was to not do the work on the device, it seems unnecessary. Intermediate heights we can handle in much the same way as now, using the sinusoidal model as listed in the NZ nautical almanac - from what I can tell, you currently calculate these on the fly anyway, so the algorithm should be reusable? Or is there any particular reason why the spring tides might be more difficult for the secondary ports?

**Comment by robinpaulson** [comment]
An intermediate step to fixing the duplication of ports problem might be to add the province to each port display name, so we get:
- French Bay, Auckland
- Waitangi, Chatham
- Waitangi, Northland
This would help no end

**Comment by robinpaulson** [comment]
From what I can gather, these are the calculations for secondary port tide depths:
H_sec = ((H_pri – MSL_pri) x ratio ) + MSL_sec
L_sec = ((L_pri – MSL_pri) x ratio ) + MSL_sec

where:
H_pri: high tide depth at primary port
L_pri: low tide depth at primary port
H_sec: high tide depth at secondary port
L_sec: low tide depth at secondary port
MSL: mean sea level
ratio: range ratio at the given secondary port

Does this look correct to you?

**Comment by jevonlongdell** [comment]
This all sounds very reasonable but I don't have time to do anything much but keep the wheels turning with NZ tides (and even that is a stretch). If you want to implement this then we should talk more. I am also open to having someone else maintain NZ tides if there is a suitable volunteer around.

**Comment by robinpaulson** [comment]
Great, I'll work that up into a series of pull requests.

I'm sort of interested in taking over maintenance, but not sure I have the confidence/experience with Java. I'd be more positive if I knew there was someone here I could ask for help when necessary.

**Comment by jevonlongdell** [comment]
I would happy to provide what help I can. I don't know Java that well myself though, NZ tides isn't a complicated program and it is the only thing I have ever written in Java.

**Comment by robinpaulson** [comment]
OK, I'll see how I go with the suggestions we've talked about. If all goes well, I'll be more keen to take it on.

**Comment by jevonlongdell** [comment]
closing this because secondary ports are already there. sure it would be nice to have them presented in a bit more of a hierachical way, but that is probably encompased in the "making ports easier to select"  issue that is still open
